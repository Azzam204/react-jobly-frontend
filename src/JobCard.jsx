import React, { useContext } from "react";
import { Button } from 'reactstrap'
import './JobCard.css'
import UserContext from "./userContext";

/** Show job's basic info
 * 
 * rendered for each job by jobslist
 * 
 * also rendered for each job or a company on company details component */ 

function JobCard({ job, applied }) {
  const { applyToJob } = useContext(UserContext)

  function handleApply(e) {
    applyToJob(job)
  }

  return (
    <div className="JobCard card">
      <div className="card-body">
        <h6 className="card-title">{job.title}</h6>
        <p>{job.companyName}</p>
        <div><small>Salary: {job.salary || 'N/A'}</small></div>
        <div><small>Equity: {job.equity || '0'}</small></div>
        {applied
          ?
          <Button disabled className="float-end apply"><b>APPLIED</b></Button>
          :
          <Button
            className="float-end apply"
            onClick={handleApply}
          ><b>APPLY</b></Button>}
      </div>
    </div>
  )
}

export default JobCard;