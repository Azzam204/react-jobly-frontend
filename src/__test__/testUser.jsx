import React from "react";
import UserContext from "../userContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
  ({ children, currUser = demoUser, isApplied = () => false }) => (
    <UserContext.Provider value={{ currUser, isApplied }}>
      {children}
    </UserContext.Provider>
  );

export { UserProvider }