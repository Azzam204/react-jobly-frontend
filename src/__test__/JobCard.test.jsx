import React from "react";
import { render } from "@testing-library/react";
import JobCard from "../JobCard";
import { UserProvider } from "./testUser";

const job = {
  title: "title",
  companyName: 'comp',
  salary: 10,
  equity: 1
}

it("renders without crashing", function () {
  render(
    <UserProvider>
      <JobCard job={job} />
    </UserProvider>
  )
})

it('matches snapshot', function () {
  const { asFragment } = render(
    <UserProvider>
      <JobCard job={job} />
    </UserProvider>
  )

  expect(asFragment()).toMatchSnapshot();
})