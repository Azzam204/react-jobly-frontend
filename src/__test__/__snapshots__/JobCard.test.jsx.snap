// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard card"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        title
      </h6>
      <p>
        comp
      </p>
      <div>
        <small>
          Salary: 10
        </small>
      </div>
      <div>
        <small>
          Equity: 1
        </small>
      </div>
      <button
        class="float-end apply btn btn-secondary"
        type="button"
      >
        <b>
          APPLY
        </b>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
